CMake message about min required cmake should be updated

**Describe the bug**

There is a status message that says Umpire needs at least cmake 3.16, but it looks like that is not enough (maybe 3.20 +?)
Building with 3.16, I see:
```
CMake Error at blt/cmake/thirdparty/BLTSetupCUDA.cmake:166 (add_library):
  add_library cannot create ALIAS target "blt::cuda" because target "cuda" is
  imported but not globally visible.
Call Stack (most recent call first):
  blt/cmake/BLTSetupTargets.cmake:107 (include)
  blt/cmake/SetupThirdParty.cmake:6 (include)
  blt/SetupBLT.cmake:129 (include)
  CMakeLists.txt:77 (include)


CMake Error at blt/cmake/thirdparty/BLTSetupCUDA.cmake:179 (add_library):
  add_library cannot create ALIAS target "blt::cuda_runtime" because target
  "cuda_runtime" is imported but not globally visible.
Call Stack (most recent call first):
  blt/cmake/BLTSetupTargets.cmake:107 (include)
  blt/cmake/SetupThirdParty.cmake:6 (include)
  blt/SetupBLT.cmake:129 (include)
  CMakeLists.txt:77 (include)
```

**Expected behavior**

Status message that says I should be required to build with cmake 3.20+ or whatever it is to remove that error above.

**Compilers & Libraries (please complete the following information):**
 - Compiler & version: [e.g. GCC 8.3.1]: 
 - CUDA version (if applicable): 11.2.0
 - Running on rzansel

**Additional context**

Add any other context about the problem here.
(This is really just to track this issue and eventually make the easy fix)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ project(Umpire
   LANGUAGES CXX C
   VERSION 2024.02.1)
 
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.23)
 
 set(UMPIRE_VERSION_RC "")
 

--- a/tests/install/using-with-cmake/CMakeLists.txt
+++ b/tests/install/using-with-cmake/CMakeLists.txt
@@ -5,7 +5,7 @@
 # SPDX-License-Identifier: (MIT)
 ##############################################################################
 
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.23)
 
 project(using_with_cmake)
 

